components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/management/WebsocketEndpointManager.java

```diff
@@ -197,12 +197,21 @@ public void closeEndpoint(int port) {
             try {
                 MessageContext synCtx = sourceHandler.getSynapseMessageContext(sourceHandler.getTenantDomain());
                 InboundEndpoint endpoint = synCtx.getConfiguration().getInboundEndpoint(endpointName);
-                shutdownStatusCode = Integer.parseInt(endpoint.getParametersMap().get("ws.shutdown.status.code"));
                 shutdownStatusMessage = endpoint.getParametersMap().get("ws.shutdown.status.message");
 
-                if (shutdownStatusCode == null) {
+                if (null != endpoint.getParametersMap().get("ws.shutdown.status.code")) {
```

Comment:
Can endpoint.getParametersMap() be null? If so, shall we do a null check for that too?

Reply:
Since we have mandatory parameters this cannot be null.